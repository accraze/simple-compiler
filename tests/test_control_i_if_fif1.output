parsing new stmt
ttype = 25
parsing Assign Statement
parsing new stmt
ttype = 25
parsing Assign Statement
parsing new stmt
ttype = 2
IF detected.... now make stmt type IFSTMT
Parsing if stmt.....
b
heres the token op1 = b
here's yer token = 20 
parsing new stmt
ttype = 25
parsing Assign Statement
parsing new stmt
ttype = 6
WE HAVE A PRINT STM
print var a
have another??
ttype 2
token IF
Exiting Statement Node
Returning Assign Statement
if_stmt->true type 102
here's yer token = 25 
Returning If Statement
Returning Assign Statement
Returning Assign Statement
3
