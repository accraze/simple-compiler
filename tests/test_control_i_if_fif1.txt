
a , b  ;
{
a = 1  ;
b = 2 ; 
IF b > a {
        a = a+b;
	print a;
	IF a > b { 
		print a;
		b = a+b;
		IF b > a { 
			print a;
			print b;
			IF a > b {
				print a;
				print b;
			}
		}
	}
	a = a+b;
	print a;
}
a = a*b;
print a;
print b;
IF a > b {
	print a;
	b = 1;
	a = a+b;
	print a;
	print b;
}

}
