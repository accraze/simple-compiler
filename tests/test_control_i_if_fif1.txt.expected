3
3
3
5
8
40
5
40
41
1
